scripts/validate-artifacts.ps1

```diff
@@ -1,121 +1,75 @@
 param(
-    [Parameter(Mandatory=$false, ValueFromPipeline=$true)]
-    [bool]$DownloadArtifacts=$true
+    [Parameter(Mandatory=$true, ValueFromPipeline=$true)]
+    [string]$ArtifactsStorageAccountName='matestorage123'
 )
 
+# default script values
+$rgName = "mate-azure-task-2"
+$taskName = "task4"
 
-# default script values 
-$taskName = "task6"
-
-$artifactsConfigPath = "$PWD/artifacts.json"
+$containerName = "task-artifacts"
+$resourcesTemplateName = "exported-template.json"
 $tempFolderPath = "$PWD/temp"
+$artifactsConfigPath = "$PWD/artifacts.json"
 
-$worksheetName = "Your Estimate"
-$spreadsheetName = "ExportedEstimate.xlsx"
-
-if ($DownloadArtifacts) { 
-    Write-Output "Reading config" 
-    $artifactsConfig = Get-Content -Path $artifactsConfigPath | ConvertFrom-Json 
-
-    Write-Output "Checking if temp folder exists"
-    if (-not (Test-Path "$tempFolderPath")) { 
-        Write-Output "Temp folder does not exist, creating..."
-        New-Item -ItemType Directory -Path $tempFolderPath
-    }
-
-    Write-Output "Downloading artifacts"
-
-    if (-not $artifactsConfig.pricingCalculationsURL) { 
-        throw "Artifact config value 'pricingCalculationsURL' is empty! Please make sure that you executed the script 'scripts/generate-artifacts.ps1', and commited your changes"
-    } 
-    Invoke-WebRequest -Uri $artifactsConfig.pricingCalculationsURL -OutFile "$tempFolderPath/$spreadsheetName" -UseBasicParsing
-
+# initial validation
+Write-Output "Running initial validation"
+$context = Get-AzContext  
+if ($context)   
+{  
+    Write-Output "Azure Powershell module is installed, account is connected."  
+} else {  
+    throw "Please log in to Azure using Azure Powershell module (run Connect-AzAccount)"
+}  
+
+Write-Output "Checking if storage account exists"
+$storageAccount = (Get-AzStorageAccount -ErrorAction SilentlyContinue | Where-Object -Property 'StorageAccountName' -EQ -Value $ArtifactsStorageAccountName )
```

Comment:
The `Get-AzStorageAccount` command is used with `-ErrorAction SilentlyContinue`. This is fine for suppressing errors, but ensure that any potential issues are logged or handled appropriately elsewhere.